src/tlo/methods/healthsystem.py

```diff
@@ -2523,6 +2567,31 @@ def record_hs_status(self, fraction_time_used_across_all_facilities: float) -> N
 
     def write_to_log_and_reset_counters(self):
         """Log summary statistics reset the data structures."""
+        
+        # If we are at the end of the year preceeding the mode switch, and if wanted to rescale capabilities to capture effective availability as was recorded, on
+        # average, in the past year, do so here.
+        # Notice that capabilities will only be expanded through this process (i.e. won't reduce available capabilities if these were under-used in the last year).
+        # Note: Currently relying on module variable rather than parameter for scale_to_effective_capabilities, in order to facilitate testing. However
+        # this may eventually come into conflict with the Switcher functions.
+        if (self.module.sim.date.year == self.module.year_mode_switch-1) and self.module.scale_to_effective_capabilities:
+
+            pattern = r"FacilityID_(\w+)_Officer_(\w+)"
+            # Calculate the average fraction of time used by officer type and level over the past year.
+            # Use len(self._frac_time_used_overall) as proxy for number of days in past year.
+            for key in self._frac_time_used_by_officer_type_and_level:
+                self._frac_time_used_by_officer_type_and_level[key] /= float(len(self._frac_time_used_overall))
+
+            for officer in self.module._daily_capabilities.keys():
+                matches = re.match(pattern, officer)
+                # Extract ID and officer type from
+                facility_id = int(matches.group(1))
+                officer_type = matches.group(2)
+                level = self.module._facility_by_facility_id[facility_id].level
+                # Only rescale if rescaling factor is greater than 1 (i.e. don't reduce available capabilities
+                # if these were under-used the previous year).
+                if self._frac_time_used_by_officer_type_and_level[officer_type + "_" + level] > 1:
+                    self.module._daily_capabilities[officer] *= \
+                        self._frac_time_used_by_officer_type_and_level[officer_type + "_" + level]
```

Comment:
This feels like something that should live in a method in `HealthSystem` module rather than here as it is largely to do with updating attributes of that module which should be the responsibility of that class, and doing it in this class requires us to store a reference to `HealthSystem` module which couples the implementations making it more difficult to test and maintain. I would say create a method of `HealthSystemSummaryCounter` to allow accessing the internal `_sum_frac_time_used_by_officer_type_and_level` as a fraction for a particular office type and level *,
```Python
    def frac_time_used_by_officer_type_and_level(self, officer_type, level):
        """Average fraction of time used by officer type and level since last reset."""
        # Use len(self._frac_time_used_overall) as proxy for number of days in past year.
        return (
            self._sum_frac_time_by_officer_type_and_level[officer_type, level] 
            / len(self._frac_time_used_overall)
        )
```
define a new internal method of `HealthSystem` module
```Python
    def _rescale_capabilities_to_capture_effective_capability(self):
        # Notice that capabilities will only be expanded through this process 
        # (i.e. won't reduce available capabilities if these were under-used in the last year).
        # Note: Currently relying on module variable rather than parameter for 
        # scale_to_effective_capabilities, in order to facilitate testing. However
        # this may eventually come into conflict with the Switcher functions.
        pattern = r"FacilityID_(\w+)_Officer_(\w+)"
        for officer in self.module._daily_capabilities.keys():
            matches = re.match(pattern, officer)
            # Extract ID and officer type from
            facility_id = int(matches.group(1))
            officer_type = matches.group(2)
            level = self.module._facility_by_facility_id[facility_id].level
            # Only rescale if rescaling factor is greater than 1 (i.e. don't reduce 
            # available capabilities if these were under-used the previous year).
            rescaling_factor = self._summary_counter.frac_time_used_by_officer_type_and_level(
                officer_type, level
            )
            if rescaling_factor > 1:
                self.module._daily_capabilities[officer] *= rescaling_factor 
```
and update the existing `on_end_of_year` method of `HealthSystem` to something like
```Python
    def on_end_of_year(self) -> None:
        """Write to log the current states of the summary counters and reset them."""
        # If we are at the end of the year preceeding the mode switch, and if wanted 
        # to rescale capabilities to capture effective availability as was recorded, on
        # average, in the past year, do so here.
        if (self.sim.date.year == self.year_mode_switch - 1) and self.scale_to_effective_capabilities:
            self._rescale_capabilities_to_capture_effective_capability()
        self._summary_counter.write_to_log_and_reset_counters()
        ...
```

*Note this also assumes that the relevant internal attribute of `HealthSystemSummaryCounter` is renamed to `_sum_frac_time_by_officer_type_and_level` as suggested above and is a `Counter` keyed by a tuple of officer type and level rather than the concatenated string, again as suggested above.